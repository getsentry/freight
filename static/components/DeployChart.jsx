import PropTypes from 'prop-types';
import React from 'react';

import createReactClass from 'create-react-class';

import api from '../api';

import BarChart from './BarChart';
import PollingMixin from '../mixins/polling';

const DeployChart = createReactClass({
  displayName: 'DeployChart',
  propTypes: {
    app: PropTypes.string,
  },
  mixins: [PollingMixin],

  getInitialState() {
    return {
      loading: true,
      points: [],
    };
  },

  componentWillMount() {
    api.request(this.getPollingUrl(), {
      success: data => {
        this.setState({
          points: this.dataToPoints(data),
          loading: false,
        });
      },
    });
  },

  getPollingUrl() {
    const {app} = this.props;
    return `/stats/${app ? `?app=${app}` : ''}`;
  },

  pollingReceiveData(data) {
    this.setState({
      points: this.dataToPoints(data),
    });
  },

  dataToPoints(data) {
    return data.map(point => {
      return {x: point[0], y: point[1]};
    });
  },

  render() {
    return (
      <div className="section">
        <BarChart points={this.state.points} label="deploys" />
      </div>
    );
  },
});

export default DeployChart;
