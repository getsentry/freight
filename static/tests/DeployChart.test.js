import React from 'react';
import {mount} from 'enzyme';

import DeployChart from '../components/DeployChart.jsx';

test('DeployChart Snapshot', () => {
  const params = {
    app: 'freight',
    env: 'production',
    number: '798',
  };
  const wrapper = mount(<DeployChart params={params} />);
  expect(wrapper).toMatchSnapshot();
});
