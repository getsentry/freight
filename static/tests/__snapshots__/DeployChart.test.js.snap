// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeployChart Snapshot 1`] = `
<DeployChart
  params={
    Object {
      "app": "freight",
      "env": "production",
      "number": "798",
    }
  }
>
  <div
    className="section"
  >
    <BarChart
      label="deploys"
      placement="bottom"
      points={Array []}
    >
      <figure
        className=" barchart"
      >
        <span
          className="max-y"
          key="max-y"
        >
          10
        </span>
        <span
          className="min-y"
          key="min-y"
        >
          0
        </span>
      </figure>
    </BarChart>
  </div>
</DeployChart>
`;
